### PR TITLE
Return DOM Name to support directing the user to better documentation

### PR DESCRIPTION
**Pull request by kasperpeulen**

### CODE REVIEW

File: lib/src/analyzer.dart [hunk 1]
```diff
@@ -163,6 +163,17 @@ class Analyzer {
               info['libraryName'] = library.name;
             }
           }
+          if (library.location.toString() == "dart:html") {
+            if (element.metadata != null) {
+              element.metadata.forEach((ElementAnnotationImpl e) {
+                if (e.toString().startsWith("@DomName")) {
+                  if (e.evaluationResult.value.fields["name"] != null) {
```
Comment (lukechurch) [inline lib/src/analyzer.dart]: Lets reduce the redundant lookup

var nameValue = e.evaluationResult.value.fields["name"];
if (nameValue != null) info["DomName"] = nameValue.value;

File: lib/src/analyzer.dart [hunk 2]
```diff
@@ -163,6 +163,17 @@ class Analyzer {
               info['libraryName'] = library.name;
             }
           }
+          if (library.location.toString() == "dart:html") {
+            if (element.metadata != null) {
+              element.metadata.forEach((ElementAnnotationImpl e) {
```
Comment (lukechurch) [inline lib/src/analyzer.dart]: It's better to do a 
for (ElementAnnotationImpl in element.metadata) here

As it means that we can add in a break; statement after we have found the DomName attribution, which short-cuts the rest of the loop.

**Comment by lukechurch** [comment]
e.evaluationResult is strictly bound to constant evaluation, so the side effects don't pose a risk.

**Comment by lukechurch** [comment]
@kasperpeulen Thanks.

I'm going to merge this in and run it through stress testing.e

File: lib/src/analyzer.dart [hunk 3]
```diff
@@ -20,6 +20,7 @@ import 'package:analyzer/src/generated/source_io.dart';
 import 'package:logging/logging.dart';
 
 import 'common.dart';
+import 'package:analyzer/src/generated/constant.dart';
```
Comment (devoncarew) [inline lib/src/analyzer.dart]: nit: import ordering

**Pull request merged**